add architecture docs (#1421)
BlueBrain/nexus: docs/src/main/paradox/docs/utilities/utilities.md

You can generate documentation using `npm run documentation` or with `docker` by running `make documentation`. More information can be found [here](https://github.com/BlueBrain/nexus-sdk-js#development).

[Source Code](https://github.com/BlueBrain/nexus-sdk-js)

Documentation:

- [nexus-sdk](https://github.com/BlueBrain/nexus-js/blob/master/packages/nexus-sdk/README.md#readme)

> ### Other Javascript Packages

Reviewer: ```suggestion
### Other JavaScript Packages
```

#### React-Nexus

> This package contains some utility components to easily integrate the Nexus SDK as React hooks or contexts

Reviewer: ```suggestion
This package contains some utility components to easily integrate the Nexus SDK as React hooks or contexts.
```

- [react-nexus](https://github.com/BlueBrain/nexus-js/blob/master/packages/react-nexus/README.md#readme)

#### Nexus-Link

> Another utility packages was written to facility chained calling behavior, which can be used independantly of Nexus.js for other projects.

Reviewer: ```suggestion
Another utility packages was written to facilitate chained calling behavior, which can be used independently of Nexus.js for other projects.
```
